crates/ethereum/primitives/src/transaction.rs

```diff
@@ -1018,8 +1021,10 @@ mod tests {
         let decoded = TransactionSigned::decode_2718(&mut raw_tx.as_slice()).unwrap();
         assert!(alloy_consensus::Typed2718::is_eip4844(&decoded));
 
-        let from = decoded.recover_signer();
-        assert_eq!(from, Some(address!("A83C816D4f9b2783761a22BA6FADB0eB0606D7B2")));
+        assert_eq!(
+            decoded.recover_signer().ok(),
+            Some(address!("95222290DD7278Aa3Ddd389Cc1E1d165CC4BAfe5"))
+        );
```

Comment:
this is a copy paste error 
```suggestion
            Some(address!("A83C816D4f9b2783761a22BA6FADB0eB0606D7B2"))
        );
```